src/firefly/js/visualize/HiPSUtil.js

```diff
@@ -0,0 +1,344 @@
+        spatialVector.set(centerWp.x, centerWp.y);
+        let radius = fov/2;
+        // we need to extend the radius
+        if (fov>60) {
+            radius *= 1.6;
+        }
+        else if (fov>12) {
+            radius *=1.45;
+        }
+        else {
+            radius *= 1.1;
+        }
+
+        pixList = hpxIdx.queryDisc(spatialVector, radius*Math.PI/180.0, true, true);
+        // add central pixel at index 0
+        const polar = Utils.radecToPolar(centerWp.x, centerWp.y);
+        ipixCenter = hpxIdx.ang2pix_nest(polar.theta, polar.phi);
+        pixList.unshift(ipixCenter);
+    }
+
+
+    let ipix;
+    for (let ipixIdx=0, len=pixList.length; ipixIdx<len; ipixIdx++) {
+        ipix = pixList[ipixIdx];
+        if (ipix===ipixCenter && ipixIdx>0) {
+            continue;
+        }
+        const cornersXYView = [];
+        const corners = HealpixCache.corners_nest(ipix, nside);
+
+        const wpCorners = [];
+        for (let k=0; k<4; k++) {
+            spVec.setXYZ(corners[k].x, corners[k].y, corners[k].z);
+            wpCorners[k] = makeWorldPt(spVec.ra(), spVec.dec(), centerWp.cSys);
+        }
+
+        cornersXYView.ipix = ipix;
+        cells.push({ ipix, wpCorners });
+    }
+
+    return cells;
+}
+
+const hipsSURVEYS = [
+    {
+        url: 'http://alasky.u-strasbg.fr/2MASS/Color',
+        label: '2MASS colored',
+    },
+    {
+        url: 'http://alasky.u-strasbg.fr/DSS/DSSColor',
+        label: 'DSS colored',
+    },
+    {
+        url: 'http://alasky.u-strasbg.fr/DSS/DSS2Merged',
+        label: 'DSS2 Red (F+R)',
+    },
+    {
+        url: 'http://alasky.u-strasbg.fr/Fermi/Color',
+        label: 'Fermi color',
+    },
+    {
+        url: 'http://alasky.u-strasbg.fr/FinkbeinerHalpha',
+        label: 'Halpha'
+    },
+    {
+        url: 'http://alasky.u-strasbg.fr/GALEX/GR6-02-Color',
+        label: 'GALEX Allsky Imaging Survey colored',
+    },
+    {
+        url: 'http://alasky.u-strasbg.fr/IRISColor',
+        label: 'IRIS colored',
+    },
+    {
+        url: 'http://alasky.u-strasbg.fr/MellingerRGB',
+        label: 'Mellinger colored',
+    },
+    {
+        url: 'http://alasky.u-strasbg.fr/SDSS/DR9/color',
+        label: 'SDSS9 colored',
+    },
+    {
+        url: 'http://alasky.u-strasbg.fr/SpitzerI1I2I4color',
+        label: 'IRAC color I1,I2,I4 - (GLIMPSE, SAGE, SAGE-SMC, SINGS)',
+    },
+    {
+        url: 'http://alasky.u-strasbg.fr/VTSS/Ha',
+        label: 'VTSS-Ha'
+    },
+    {
+        url: 'http://saada.u-strasbg.fr/xmmallsky',
+        label: 'XMM-Newton stacked EPIC images (no phot. normalization)',
+    },
+    {
+        url: 'http://saada.u-strasbg.fr/xmmallsky/',
+        label: 'XMM PN colored',
+    },
+    {
+        url: 'http://alasky.u-strasbg.fr/AllWISE/RGB-W4-W2-W1/',
+        label: 'AllWISE color',
+    },
+    {
+        url: 'http://www.spitzer.caltech.edu/glimpse360/aladin/data',
+        label: 'GLIMPSE360',
+    }
+];
+
```

Comment:
Just curious, how did we come with this list of surveys? 

Is it possible to filter them from the list of available surveys dynamically?

http://aladin.u-strasbg.fr/java/nph-aladin.pl?frame=aladinHpxList gives this link:

http://alasky.unistra.fr/MocServer/query?hips_service_url=*&dataproduct_type=!catalog&dataproduct_type=!cube&get=record 

and `hips_service_url` gives a URL of the image service.

Reply:
I plan to make it more dynamic in another ticket.